### DISCUSSION TITLE
API Documentation:  to generate everything except transcode

### DISCUSSION THREAD

**Discussion by lamyergeier** [stashapp/stash#4720] [Q&A] [answered]
- I checked [API  Stash-Docs](https://docs.stashapp.cc/networking/api/)
- Following works (as described in above url)

```bash
curl -X POST -H "Content-Type: application/json" --data \
    '{ "query": "mutation { metadataGenerate ( input : { sprites: false previews: true imagePreviews: false markers: false transcodes: false } ) }" }' \
    localhost:9998/graphql
```

But I would like to generate everything as per  following image

![image](https://github.com/stashapp/stash/assets/42092626/d82c4ee1-e583-400e-9771-ae776ead832f)

How to find API key for each of above?

**Comment by DogmaDragon** [comment]
Go to `/playground` to see the full schema. E.g. `http://localhost:9999/playground`
![image](https://github.com/stashapp/stash/assets/103123951/17075322-1ddc-459d-8009-5681f8cca86c)

**Comment by stg-annon** [comment, marked as answer]
this is what you would be looking for is [GenerateMetadataInput](https://github.com/stashapp/stash/blob/develop/graphql/schema/types/metadata.graphql#L1-L25)

a useful API reference can be found at `localhost:9999/playground` this is dynamically generated and will be up to date with whatever version you are running
